src/org/exist/source/SourceFactory.java

```diff
@@ -163,6 +158,26 @@ private static String firstPathSegment(final String path) {
                 .getRawCollectionPath();
     }
 
+    private static Source getSource_fromClasspath(final String contextPath, final String location) throws IOException {
+        if (location.startsWith(ClassLoaderSource.PROTOCOL)) {
+            return new ClassLoaderSource(location);
+        }
+
+        final Path rootPath = Paths.get(contextPath.substring(ClassLoaderSource.PROTOCOL.length()));
```

Comment:
this will resolve a *file* on literal `/path/to/file` correct? is this intended? does this mean that if I would write `resource:/etc/shadow` on a linux box that I would see all users and password hashes?

Reply:
No. `resource:` limits the resolution to the classpath. So you can't resolve things which are not already on the classpath. The `Path` class is just used to make it easier to manipulate the paths for the classpath correctly.